### ISSUE TITLE
First note muted

### ISSUE THREAD

**Issue by TeamB12** [rncbc/qtractor#446]
Sometimes, the first note in a clip is not played.
Double checked everything: switching plugin instrument will not solve the problem.
Notes not being played is not so urgent. Qtractor is crashing from time to time without no reason (after completing plugin scan) - this is much more urgent, see my previous "bug report", here.
Let's not expect too much  ( or anything at all) and we will not be dissapointed.
Main problem concerning Qtractor is stability - the dev is surely doing his best.

**Comment by rncbc** [comment]
like the others you issued, this is a too vague of a bug report;

please tell which versions, builds, packaging where did you get qtractor from, if not built from source, etc.

also, try to isolate the symptoms into a minimal session and post it for the issue to be reproduced (please save as .qtz to have all audio/midi files bundled accordingly; avoid any copyrighted material).

**Comment by TeamB12** [comment]
OS:Artix Linux,  Init system: OpenRC, Kernel: 6.7.0.6.realtime1-4-rt
Xorg, Jack1,  Jack Audio Buffers: n1 p128
Qtractor Version 1.0  binary package from package manager.
Issue: first not gets deleted - sorryy, I thought first it was muted, but those notes seemed to be removed.
The first note of a clip gets sometimes removed, this is a Bug Report!
Different "things" seem to trigger the Bug - I ve recorded a Midi clip, then I saved and after that the first note was removed. So, just saving can trigger a Qtractor Bug that deletes the first note. I wonder why no one else has reported that.
Summary: Qtractor 1.0 from Arch Linux Repository still shows this bug.
If I am still too vague, please let me know and I will happily give all missing information.

**Comment by rncbc** [comment]
can you screencast the issue into a video or animated gif, please?
what's the input source of the MIDI recording? a MIDI keyboard controller or something else? how it is connected to qtractor? alsa-midi, jack-midi, pipewire ?

**Comment by TeamB12** [comment]
There is one usb midi keyboard connected - I have connected it with qjackctl.
In Qjackctl I have selected no midi driver, so I hope its jack midi, that is used concerning midi.
No pipewire, no pulse - I mean there are maybe libpipewire or libpulse installed for dependency reasons, but those are not started.
Of course I can make a video - but you won't see more then I decribed.
Little Problem, the first note removed problem is occurring  quite rarely - but I will try.y

**Comment by TeamB12** [comment]
Finally I got some Qtractor bugs caught on video.
You can watch em here:
[https://www.youtube.com/watch?v=szswZLuEs-Q](url)
https://www.youtube.com/watch?v=szswZLuEs-Q

Video is 9 minutes long - I am sorry, but, those bugs and issues are really nasty and random.
Making a video felt strange and that I am the first who reports those bugs feels even stranger.
Just watch the last minute, and then the beginning of the video and you can directly see whats going on.
The issues again:
First notes in several midi clips in Qtractor are suddenly missing.
Sometimes, notes are shifted.

Linux is so powerful because everything is perfectly isolated, like the GUI.
How much work would it be to port Qtractor to GTK3 ?

Right now, as video is proofing,I can not work with Qtractor.
Fixing all those strange missing notes and wrong notes allways takes maybe 15 minutes, and then, some minutes later all thoses wrong notes are back again.
Long awaited version 1 seems one of the less reliable Qtractor versions.

**Comment by rncbc** [comment]
I hear your call of pain and despair :)

while having (a lot) of investigation I can now swear the culprit has been found: the vanishing notes appears on MIDI clips that were head-cut from the original full-length MIDI file, thus having a positive non-zero offset. and notes are interted at the start as first notes.

problem is due to rounding errors alone and might not be noticeable in a deterministic way... iow. doesn't happen on some situations while it does on others, so sorry.

afaict. the "bug" has been there lurking for quite some time, prolly introduced somewhere after v0.9.28, almost 2 years ago.

all that to tell that a regression has been taken and qtractor >= v1.0.0.13git.[b4fb6b](https://github.com/rncbc/qtractor/commit/b4fb6b0) might have nailed it.

so please, play it again Sam ;)

cheers

ps.
> How much work would it be to port Qtractor to GTK3 ?

are you kidding? sure you are :)

**Comment by TeamB12** [comment]
Thank you.
So, I am a user not a dev, let me understand it: 
the "issue" has been there for 2 years and it is still there in version 1 and it will be there for some time.
Concerning my music project: I have to live with it?
Sorry, I wasted your time and my time.
My personal view: some small rough edges are no problem, but with this "issue", it is really hard to make music.
Maybe I am stupid, I do not see a "pre roll" like in Cubase:
my project is in the loop and the loop starts at bar 5, so I have 4 pre counts...
If I understod you correctly, the way, I am working is the problem:
after recording, I have to cut or resite to get a clip that starts at 5. The, I will repat it...

**Comment by rncbc** [comment]
> the "issue" has been there for 2 years and it is still there in version 1 and it will be there for some time.

yes and no, the issue is believed to be fixed on v1.0.0.13 onwards.

> I do not see a "pre roll" like in Cubase:

it's called "count in" in qtractor; it's a number of beats (quarter-notes)  that plays on hold before playback or recording actually starts rolling; check View > Options... > Audio (or MIDI) > Metronome, for configuration.

> after recording, I have to cut or resize to get a clip that starts at 5. Then, I will repeat it...

yes, the "bug" manifests more probably on those clips that you cut or resize their head (4 bars in your case).

**Comment by TeamB12** [comment]
Thanks again.
A good small tutroail about debugging on the landing page of the "support/FAQ" section of Qtractor's home page would be a good starting point for normal users like me.
Maybe, a ready to install Qtractor-debug binary package in the Linux distribution's package manager would be really comfortable.
Ontopic: it seems, the issue happens while or after saving the project.
The problem doubled here because, each time I am moving or editing my linked midi clip, Qtractor wants to save - I guess, that is not an automatic save feature. 
I have not readt the manual yet, my fault.
thanks again.

**Comment by rncbc** [comment]
> A good small tutroail about debugging on the landing page of the "support/FAQ" section of Qtractor's home page would be a good starting point for normal users like me.

there's always the official [wiki](https://sourceforge.net/p/qtractor/wiki/Home), reflected to the official user manuals in https://qtractor.org/doc; though all of these are or have been authored by users like you, so it's either way incomplete and/or severly outdated, specially regarding some _niche_ topics and features.

> Maybe, a ready to install Qtractor-debug binary package in the Linux distribution's package manager would be really comfortable.

please check the personal [repos](https://www.rncbc.org/drupal/node/2145); there you'll find the most recent snapshot builds for the most popular distro packagings.

> Ontopic: it seems, the issue happens while or after saving the project. The problem doubled here because, each time I am moving or editing my linked midi clip, Qtractor wants to save - I guess, that is not an automatic save feature. I have not readt the manual yet, my fault. thanks again.

yes exactly, as said earlier:
>> notes that are inserted at the start of the clip are missed upon reload--the notes are indeed written to the file, just don't get read into the clip mostly due to rounding errors.

that meaning--for x-sake--that the _disappearing_ notes are still there, in the file, just not get read on reload :/ one side-effect of this happening will be evident when you get to run a fixed build (>= v1.0.0.13), so those hidden notes will appear magically there back again :)

cheers

**Comment by TeamB12** [comment]
got it now ;)
thank you - I will wait for Qtractor ver >= v1.0.0.13.
I will give a short remark here, when the vanished notes are back again.

**Comment by TeamB12** [comment]
Sorry for the inconveniences:
Using the pre count feature now and I have notcut or deleted any clip - still the same problem.
No exactly the same: what is the dame is, that I can not hear the first note in a clip.
Whats new now is, that I can see the note.
I ve made a video - sigh.
Pre count is good, but still the same problem: sometimes I have to cut clips.
And I am using automation to get everything right in the mix session.
Normally, in the first seconds I am  setting the volumes of each track by using the automation.
Now, I am starting at bar 1 and the automation has not time to get everything right before the first note starts.
Mmh, I need pre count, and then I must start recording at bar 5.
If you want to see the video, I ve uploaded it:

**Comment by rncbc** [comment]
> so please, play it again Sam ;)

meant after the Steely Dan's _Do It Again_ (Jack), not that casablanca thing, sorry :)

**Comment by TeamB12** [comment]
Maybe I am seeing ghosts - probably, everything is ok now, maybe the last note is hanging, but that is also not a Qtractor specific issue.
https://www.youtube.com/watch?v=Jr4j1cSzTJI
